hedera-node/src/main/java/com/hedera/services/txns/consensus/TopicUpdateTransitionLogic.java

```diff
@@ -55,247 +55,151 @@
 		}
 	}
 
-	private void applyNewAutoRenewPeriod(ConsensusUpdateTopicTransactionBody op, MerkleTopic topic) {
+	private void validateNewAutoRenewPeriodApplicability(final ConsensusUpdateTopicTransactionBody op) {
 		if (op.hasAutoRenewPeriod()) {
-			topic.setAutoRenewDurationSeconds(op.getAutoRenewPeriod().getSeconds());
-		}
-	}
-
-	private boolean designatesAccountRemoval(AccountID id) {
-		return id.getShardNum() == 0 && id.getRealmNum() == 0 && id.getAccountNum() == 0;
-	}
-
-	private boolean canApplyNewKeys(ConsensusUpdateTopicTransactionBody op, MerkleTopic topic) {
-		if (!op.hasAdminKey() && !op.hasSubmitKey()) {
-			return true;
+			validateFalse(!validator.isValidAutoRenewPeriod(op.getAutoRenewPeriod()), AUTORENEW_DURATION_NOT_IN_RANGE);
 		}
-		var topicId = op.getTopicID();
-		try {
-			if (op.hasAdminKey()  && !applyNewAdminKey(op, topicId, topic)) {
-				return false;
-			}
-
-			if (op.hasSubmitKey()  && !applyNewSubmitKey(op)) {
-				return false;
-			}
-		} catch (DecoderException e) {
-			log.error("Decoder exception updating topic {}. ", topicId, e);
-			transactionContext.setStatus(BAD_ENCODING);
-			return false;
-		}
-		return true;
 	}
 
-	private boolean applyNewSubmitKey(final ConsensusUpdateTopicTransactionBody op) throws DecoderException {
-		var newSubmitKey = op.getSubmitKey();
-		if (!validator.hasGoodEncoding(newSubmitKey)) {
-			transactionContext.setStatus(BAD_ENCODING);
-			return false;
-		}
-		JKey.mapKey(newSubmitKey);
-		return true;
-	}
+	/*
+	IMPORTANT:
+	The CryptoDeleteTransitionLogicRefactoring PR introduces new implementation of the private validateUsable() method,
+	executed when AccountStore.loadAccount() is being called.
 
-	private boolean applyNewAdminKey(final ConsensusUpdateTopicTransactionBody op,
-			final TopicID topicId,
-			final MerkleTopic topic) throws DecoderException {
-		var newAdminKey = op.getAdminKey();
-		if (!validator.hasGoodEncoding(newAdminKey)) {
-			log.error("Update topic {} has invalid admin key specified, " +
-					"which should have been caught during signature validation", topicId);
-			transactionContext.setStatus(BAD_ENCODING);
-			return false;
-		}
-		var fcKey = JKey.mapKey(newAdminKey);
-		if (fcKey.isEmpty()) {
-			boolean opRemovesAutoRenewId = op.hasAutoRenewAccount() &&
-					designatesAccountRemoval(op.getAutoRenewAccount());
-			if (topic.hasAutoRenewAccountId() && !opRemovesAutoRenewId) {
-				transactionContext.setStatus(AUTORENEW_ACCOUNT_NOT_ALLOWED);
-				return false;
+	After merge of this PR with the Hedera Master branch - the redundant currentAutoRenewAccount.isSmartContract() validation
+	should be removed - it will be executed by validateUsable().
+	*/
+	private void validateNewAutoRenewAccountApplicability(final ConsensusUpdateTopicTransactionBody op,
+														  final Topic topic,
+														  final AccountStore accountStore) {
+		if (op.hasAutoRenewAccount()) {
+			if (!designatesAccountRemoval(op.getAutoRenewAccount())) {
```

Comment:
Instead of introducing new method `designatesAccountRemoval`, why don't we use `AccountID.newDefaultInstance`?